docs/entrance randomization.md

```diff
@@ -0,0 +1,377 @@
+# Entrance Randomization
+
+This document discusses the API and underlying implementation of the generic entrance randomization algorithm
+exposed in [entrance_rando.py](/entrance_rando.py). Throughout the doc, entrance randomization is frequently abbreviated
+as "ER."
+
+This doc assumes familiarity with Archipelago's graph logic model. If you don't have a solid understanding of how
+regions work, you should start there.
+
+## Entrance randomization concepts
+
+### Terminology
+
+Some important terminology to understand when reading this doc and working with ER is listed below.
+
+* Entrance rando - sometimes called "room rando," "transition rando," "door rando," or similar,
```

Comment:
My understanding was that it is correct grammar to have it inside, apparently that is not true however a quick google for "does comma go inside or outside quotes" brings back plenty of reputable sources recommending inside, at least in American English. Here is a secondary source with this exact example, even: https://ell.stackexchange.com/questions/257851/should-the-comma-be-inside-or-outside-of-the-quotation-marks

I guess this raises the question though, is there an actual preferred style guide for docs? If not there should be (the opportunity to use AP style is very funny to me).

Reply:
I think the comma is inside the quotes only if it is used for a quote - i.e. someone saying something. Here the quotes are used to mark an expression. I could be wrong as non-native speaker.

(I would go for `"room rando", ...`)

Reply:
After googling for a bit, I came to the conclusion that it depends on AmE vs. BrE:
https://www.reddit.com/r/grammar/comments/11c3aeg/when_using_scare_quotes_does_punctuation_go/

While this breaks my programmer brain, I think it would be correct since we mostly use AmE.

Reply:
This is kind of just another source for what you found, but I wanted to clarify that I came to the same conclusion, and that I think we'll go with AmE style until decided otherwise.